demo-app-benchmark-test/src/main/java/cash/z/ecc/android/sdk/demoapp/benchmark/StartupBenchmark.kt

```diff
@@ -0,0 +1,111 @@
+        private const val APP_TARGET_PACKAGE_NAME = "cash.z.ecc.android.sdk.demoapp.mainnet"    // NON-NLS
+
+        private const val ADDRESS_SCREEN_SECTION = "ADDRESS_SCREEN" // NON-NLS
+        private const val UNIFIED_ADDRESS_SECTION = "UNIFIED_ADDRESS"   // NON-NLS
+        private const val SAPLING_ADDRESS_SECTION = "SAPLING_ADDRESS"   // NON-NLS
+        private const val TRANSPARENT_ADDRESS_SECTION = "TRANSPARENT_ADDRESS"   // NON-NLS
+    }
+
+    @get:Rule
+    val benchmarkRule = MacrobenchmarkRule()
+
+    /**
+     * This test starts the Demo-app on Home screen and measures its metrics.
+     */
+    @Test
+    fun appStartup() = benchmarkRule.measureRepeated(
+        packageName = APP_TARGET_PACKAGE_NAME,
+        metrics = listOf(StartupTimingMetric()),
+        iterations = 5,
+        startupMode = StartupMode.COLD,
+        setupBlock = {
+            // Press home button before each run to ensure the starting activity isn't visible
+            pressHome()
+        }
+    ) {
+        startActivityAndWait()
+    }
+
+    /**
+     * Advanced trace events startup test, which starts the Demo-app on the Home screen and then navigates to the
+     * Get Address screen. Logic for providing addresses from SDK is measured here.
+     */
+    @Test
+    @OptIn(ExperimentalMetricApi::class)
+    fun tracesSdkStartup() = benchmarkRule.measureRepeated(
+        packageName = APP_TARGET_PACKAGE_NAME,
+        metrics = listOf(
+            TraceSectionMetric(sectionName = ADDRESS_SCREEN_SECTION, mode = TraceSectionMetric.Mode.First),
+            TraceSectionMetric(sectionName = UNIFIED_ADDRESS_SECTION, mode = TraceSectionMetric.Mode.First),
+            TraceSectionMetric(sectionName = SAPLING_ADDRESS_SECTION, mode = TraceSectionMetric.Mode.First),
+            TraceSectionMetric(sectionName = TRANSPARENT_ADDRESS_SECTION, mode = TraceSectionMetric.Mode.First)
+        ),
+        compilationMode = CompilationMode.Full(),
+        startupMode = StartupMode.COLD,
+        iterations = 5,
+        measureBlock = {
+            startActivityAndWait()
+            gotoAddressScreen()
+            waitForAddressScreen()
+            closeAddressScreen()
+        }
+    )
+
+    private fun MacrobenchmarkScope.closeAddressScreen() {
+        // To close the Address screen and disconnect from SDK Synchronizer
+        device.pressBack()
+    }
+
+    private fun MacrobenchmarkScope.waitForAddressScreen() {
+        val timeoutSeconds = 5.seconds
+        check(
+            device.run {
+                val ua = Until.hasObject(By.text("^[a-z0-9]{141}$".toPattern())) // NON-NLS (Unified address condition)
+                val sa = Until.hasObject(By.text("^[a-z0-9]{78}$".toPattern())) // NON-NLS (Sapling address condition)
+                val ta = Until.hasObject(By.text("^[a-zA-Z0-9]{35}$".toPattern())) // NON-NLS (Transparent address)
+                // condition
+
+                wait(ua, timeoutSeconds.inWholeMilliseconds) != null &&
```

Comment:
Would it be helpful to add a few global extension functions for `wait` and `clickAndWait` which accept `Duration` so that we don’t have to call `.inWholeMilliseconds` everywhere?

Reply:
Yep, thanks for pointing this out. It looks much better now.